### PR TITLE
Refactor exception classes and exception handling 

### PR DESCRIPTION
**Pull request by yardenvan**
1. Removed errorMsg variable and added use of flogger formatting
2. Added NullPointerException handling in asset creation in case not all fields exist in their property map
3. Removed exceptions folder and created one class with all of the exceptions as static nested classes

### CODE REVIEW

File: src/main/java/com/google/cloudassets/discovery/AssetDiscoveryExceptions.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            super(errorMsg, cause);
+        }
+    }
+
+    /**
+     * This class extends the Exception class and should be thrown when no table is configured for a
+     * specific asset kind in our configuration table.
+     */
+    public static class NoTableConfigException extends Exception {
+        public NoTableConfigException() {
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/AssetDiscoveryExceptions.java]: I think you can simply not define a constructor in this case. The predefined default constructor will do the same thing.

File: src/main/java/com/google/cloudassets/discovery/AssetDiscoveryExceptions.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+    }
+
+    /**
+     * This class extends the Exception class and should be thrown when too many tables are configured
+     * for a specific asset kind in our configuration table.
+     */
+    public static class TooManyTablesConfigException extends Exception {
+        public TooManyTablesConfigException() {
+            super();
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/AssetDiscoveryExceptions.java]: also here

File: src/main/java/com/google/cloudassets/discovery/assetobjects/AppAppEngineObject.java [hunk 3]
```diff
@@ -49,18 +49,23 @@ public Builder(Map<String,Object> assetProperties, ProjectConfig projectConfig)
...
-            specificObjectClass.databaseType = castToString(assetProperties.get("databaseType"));
+                // Set specific asset type fields
+                specificObjectClass.authDomain = castToString(assetProperties.get("authDomain"));
+                specificObjectClass.defaultHostname = castToString(assetProperties.get("defaultHostname"));
+                specificObjectClass.codeBucket = castToString(assetProperties.get("codeBucket"));
+                specificObjectClass.gcrDomain = castToString(assetProperties.get("gcrDomain"));
+                specificObjectClass.defaultBucket = castToString(assetProperties.get("defaultBucket"));
+                specificObjectClass.databaseType = castToString(assetProperties.get("databaseType"));
+            } catch (NullPointerException exception) {
+                logger.atInfo().withCause(exception).log("Could not set all of the AppAppEngineObject " +
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/assetobjects/AppAppEngineObject.java]: If you define your own "get" method, the exception message will be able to include the exception include the name of the missing property and also make this method shorter and more readable.

File: src/main/java/com/google/cloudassets/discovery/assetobjects/BucketStorageObject.java [hunk 4]
```diff
@@ -47,15 +47,19 @@ public Builder(Map<String,Object> assetProperties, ProjectConfig projectConfig)
...
+            try {
+                setName(assetProperties.get("name"));
+                setId(assetProperties.get("id"));
+                setLocation(getLastSeg(assetProperties.get("location")));
+                setCreationTime(convertStringToDate(assetProperties.get("timeCreated")));
 
+                // Set specific asset type fields
+                specificObjectClass.storageClass = castToString(assetProperties.get("storageClass"));
+                specificObjectClass.updatedTime = convertStringToDate(assetProperties.get("updated"));
+            } catch (NullPointerException exception) {
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/assetobjects/BucketStorageObject.java]: also here and in a few other places below

**Review by itaitz — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/cloudassets/discovery/assetobjects/AppAppEngineObject.java (hunk 3 above)
Comment (yardenvan) [reply src/main/java/com/google/cloudassets/discovery/assetobjects/AppAppEngineObject.java]: That is a good idea but the problem is that the get method does not throw the NullPointerException it just returns null and then the function that uses this value throws the NullPointerException.
I can add a try & catch statement to helper functions defined in the AssetObject class if you think it makes more sense (but then we still do not have the name of the missing property). 
WDYT?

**Review by itaitz — APPROVED** [review]

**Pull request merged**